sdk/iot/hub/tests/cmocka/test_az_iot_hub_client_c2d.c

```diff
@@ -204,29 +189,66 @@ static void test_az_iot_hub_client_c2d_parse_received_topic_URL_ENCODED_succeed(
 
   az_iot_hub_client_c2d_request out_request;
 
-  assert_return_code(
+  assert_int_equal(
       az_iot_hub_client_c2d_parse_received_topic(&client, received_topic, &out_request), AZ_OK);
 
   az_pair pair;
-  assert_return_code(az_iot_hub_client_properties_next(&out_request.properties, &pair), AZ_OK);
+  assert_int_equal(az_iot_hub_client_properties_next(&out_request.properties, &pair), AZ_OK);
   assert_true(az_span_is_content_equal(pair.key, AZ_SPAN_FROM_STR("%24.to")));
   assert_true(az_span_is_content_equal(pair.value,
   AZ_SPAN_FROM_STR("%2Fdevices%2Fuseragent_c%2Fmessages%2FdeviceBound")));
   
-  assert_return_code(az_iot_hub_client_properties_next(&out_request.properties, &pair), AZ_OK);
+  assert_int_equal(az_iot_hub_client_properties_next(&out_request.properties, &pair), AZ_OK);
   assert_true(az_span_is_content_equal(pair.key, AZ_SPAN_FROM_STR("abc")));
   assert_true(az_span_is_content_equal(pair.value, AZ_SPAN_FROM_STR("123")));
 
-  assert_return_code(az_iot_hub_client_properties_next(&out_request.properties, &pair), AZ_OK);
+  assert_int_equal(az_iot_hub_client_properties_next(&out_request.properties, &pair), AZ_OK);
   assert_true(az_span_is_content_equal(pair.key, AZ_SPAN_FROM_STR("ghi")));
   assert_true(az_span_is_content_equal(pair.value, AZ_SPAN_FROM_STR("%2Fsome%2Fthing")));
 
-  assert_return_code(az_iot_hub_client_properties_next(&out_request.properties, &pair), AZ_OK);
+  assert_int_equal(az_iot_hub_client_properties_next(&out_request.properties, &pair), AZ_OK);
   assert_true(az_span_is_content_equal(pair.key, AZ_SPAN_FROM_STR("jkl")));
   assert_true(az_span_is_content_equal(pair.value,
   AZ_SPAN_FROM_STR("%2Fsome%2Fthing%2F%3Fbla%3Dbla")));
 }
 
+static void test_az_iot_hub_client_c2d_parse_received_topic_fail(void** state)
```

Comment:
Nit: Naming should be updated to `unit_action_result`. E.g. c2d_parse_received_topic_unrelated_topic_reject. (This is part of normal operation so not a failure.)

Reply:
good point. updated